From a372edcb73e50bb2f1d2c805a83d6c2fa92afcad Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@st.com>
Date: Thu, 6 Apr 2023 11:35:05 +0200
Subject: [PATCH 483/529] net: ethernet: stmmac: encapsulate power managements
 functions

This commit encapsulates all the power management functions.

Signed-off-by: Arturo Buzarra <arturo.buzarra@digi.com>
Signed-off-by: Christophe Roullier <christophe.roullier@st.com>
Change-Id: I94d6e73936e718bdf7b81c35bf439bc34380501b
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/299335
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Alexandre TORGUE <alexandre.torgue@foss.st.com>
Domain-Review: Alexandre TORGUE <alexandre.torgue@foss.st.com>
---
 drivers/net/ethernet/stmicro/stmmac/dwmac-stm32.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/net/ethernet/stmicro/stmmac/dwmac-stm32.c
+++ b/drivers/net/ethernet/stmicro/stmmac/dwmac-stm32.c
@@ -500,10 +500,10 @@ static int stm32_dwmac_resume(struct dev
 
 	return ret;
 }
-#endif /* CONFIG_PM_SLEEP */
 
 static SIMPLE_DEV_PM_OPS(stm32_dwmac_pm_ops,
 	stm32_dwmac_suspend, stm32_dwmac_resume);
+#endif /* CONFIG_PM_SLEEP */
 
 static struct stm32_ops stm32mcu_dwmac_data = {
 	.set_mode = stm32mcu_set_mode,
@@ -533,7 +533,9 @@ static struct platform_driver stm32_dwma
 	.remove = stm32_dwmac_remove,
 	.driver = {
 		.name           = "stm32-dwmac",
+#ifdef CONFIG_PM_SLEEP
 		.pm		= &stm32_dwmac_pm_ops,
+#endif /* CONFIG_PM_SLEEP */
 		.of_match_table = stm32_dwmac_match,
 	},
 };
